### ISSUE TITLE
Nuget gathething folder is empty for .NET 6.0 Preview 1

### ISSUE THREAD

**Issue by rbhanda** [dotnet/arcade#6957]
Copy to nuget should happen during Arcade SDK, however the folder is empty.
We have to:

1. Investigate why the copy did not work
2. Ensure that the signed bits are updated in publish_files prior to this copy when post-build sign = true

**Comment by riarenas** [comment]
@rbhanda trying to understand the urgency here:

1. Is this an ask for dnceng to look into this? If so, can you elaborate more details on what you mean by copy to nuget, and which folder you're talking about? Is there a link to something we can look at?
1. Is this blocking something? Do we need to address it right away? (using the default issue template in this repo would be best for this)

**Comment by rbhanda** [comment]
@mmitche has helped to resolved this issue. 
This issue is created to investigate why nuget folder under the release drop was not populated for upcoming release.

**Comment by riarenas** [comment]
Oh if @mmitche is involved already we're good. Was just afraid of this getting lost until we triaged it next week.

**Comment by adiaaida** [comment]
The second part will need to be addressed outside of gather-drop.

Also, I think the reason that this wasn't populated correctly is that there may be a difference between what gather-drop produces and what initialize release expects. Gather-drop dropped the preview bits here:

```\\vsufile\patches\sign\NET\CORE_BUILDS\6.0.X\6.0.0-preview.1\6.0.100-preview.1.21103.13\publish_files```

but initialize-release looked for them here: 

```\\vsufile\patches\sign\NET\CORE_BUILDS\6.0.X\6.0.0\6.0.100-preview.1.21103.13\publish_files```

So there is probably something that needs to be fixed in this code to make it work correctly for preview builds: https://dev.azure.com/dnceng/internal/_git/dotnet-release?path=%2Fsrc%2Finitialize-release%2Fsrc%2FFilesToRelease.cs&version=GBmain&line=32&lineEnd=33&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents.

Or gather-drop needs to drop things in a different place.

**Comment by adiaaida** [comment]
I looked into this further, and this doesn't even seem to be a gather-drop failure, but a human error. Basically, when we ran the pipeline, we told it to drop the files into:

```release-output-dir : \\vsufile\patches\sign\NET\CORE_BUILDS\6.0.X\6.0.0-preview.1\6.0.100-preview.1.21103.13```

But, as stated above, that's not where initialize-release expects them. The fix for this is to make sure the paths that we tell the pipeline to drop files matches what is expected by the tools later down the line. However, even better is that this will be solved by the work to remove vsufile from the inner loop of the release pipeline, as we will be explicitly telling the tools where to look for files. I will confirm this today with my test run on my branch.
